src/test/java/hudson/plugins/timestamper/pipeline/PipelineTest.java

```diff
@@ -0,0 +1,143 @@
+package hudson.plugins.timestamper.pipeline;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.gargoylesoftware.htmlunit.WebClientUtil;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.gargoylesoftware.htmlunit.html.HtmlPreformattedText;
+import com.gargoylesoftware.htmlunit.html.HtmlSpan;
+import hudson.plugins.timestamper.TimestamperConfig;
+import java.io.BufferedReader;
+import java.io.StringReader;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class PipelineTest {
+
+    private boolean originalAllPipelines;
+
+    @Rule public JenkinsRule r = new JenkinsRule();
+
+    @Before
+    public void setAllPipelines() {
+        TimestamperConfig config = TimestamperConfig.get();
+        originalAllPipelines = config.isAllPipelines();
+        config.setAllPipelines(true);
+    }
+
+    @After
+    public void restoreAllPipelines() {
+        TimestamperConfig config = TimestamperConfig.get();
+        config.setAllPipelines(originalAllPipelines);
```

Comment:
This is unnecessary. The setting is already thrown away at the end of every test case by `JenkinsRule`.